### PR TITLE
added check of _is_variable

### PR DESCRIPTION
**Pull request by usama-baloch**
updated the behaviour of all torch backend functions with support_native_out=True, using the _is_variable in handle_out_argument decorator.

### CODE REVIEW

File: ivy/func_wrapper.py [hunk 1]
```diff
@@ -847,6 +848,10 @@ def _handle_out_argument(*args, out=None, **kwargs):
         """
         if out is None or is_compos_fn:
             return fn(*args, out=out, **kwargs)
+
+        if _is_variable(out):
```
Comment (VedPatwardhan) [inline ivy/func_wrapper.py:852]: probably we can do a `handle_out_in_backend = handle_out_in_backend and _is_variable(out)` instead as that's a bit cleaner

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @usama-baloch, thanks for making the changes, just one minor change requested 🙂

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @usama-baloch, one last change and we're good to merge 🙂

File: ivy/func_wrapper.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 from types import FunctionType
 from typing import Callable
 import inspect
+from ivy.functional.ivy.gradients import _is_variable
```
Comment (VedPatwardhan) [inline ivy/func_wrapper.py:11]: Let's remove this import as it leads to a circular import error

File: ivy/func_wrapper.py [hunk 3]
```diff
@@ -847,6 +848,7 @@ def _handle_out_argument(*args, out=None, **kwargs):
         """
         if out is None or is_compos_fn:
             return fn(*args, out=out, **kwargs)
+        handle_out_in_backend = handle_out_in_backend and not _is_variable(out)
```
Comment (VedPatwardhan) [inline ivy/func_wrapper.py:851]: Given that we're not able to import `_is_variable` in this file, let's just move back to the way you were implementing it,
```
        if ivy.gradients._is_variable(out):
            handle_out_in_backend = False
```

**Review by VedPatwardhan — CHANGES_REQUESTED** [review]
Hey @usama-baloch, just requested a couple of changes, we're ready to merge once they're done. Thanks 🙂

File: ivy/func_wrapper.py (hunk 3 above)
Comment (usama-baloch) [reply ivy/func_wrapper.py:851]: Sure 👍

**Pull request merged**